File: Models/Order.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
namespace ServerVKR.Models
{
    public class Order
    {
        public Guid Id { get; set; } = Guid.NewGuid();
        public string FirstName { get; set; }
        public string LastName { get; set; }
        public string Comment { get; set; }
        public string Phone { get; set; }
        public string Address { get; set; }
        public List<OrderItem> OrderItems { get; set; }
        public Delivery Deliverys { get; set; }
        public int TotalPtice { get; set; }
        public DateTime CreatedDate { get; set; }
        public DateTime DeliveryDate { get; set; }
        public string Status { get; set; }
        public int Number { get; set; }
        
    }
}

File: wwwroot/js/order.js
$(document).ready(function () {
    $('#checkout').on('click', function () {
        let order = {
            firstName: $("#firstname").val(),
            lastName: $("#lastname").val(),
            comment: $("#comment").val(),
            phone: $("#phone").val(),
            address: $("#address").val(),
            deliveryDate: $("#delivery-date").val(),
            orderItems: [],
            deliverys: {
                id: $("#delivery-method").val()
            }
        }

        let cart = JSON.parse(localStorage.getItem('cart'))


        cart.items.forEach(item => {
            order.orderItems.push({
                product: {
                    id: item.id
                },
                quantity: item.quantity,
                price: item.price
            })
        })

        console.log(order)

        localStorage.removeItem('cart')


        $.post("/Home/Checkout", order).done(function (data) {
            document.location.href = "/"
        });

    })

    let cart = JSON.parse(localStorage.getItem('cart'))
    let totalPrice = 0

    cart.items.forEach(item => {
        totalPrice += item.price*item.quantity
    })

    $('#total-price').html(totalPrice)
})